src/collections/PricesCollection.php

```diff
@@ -134,9 +127,24 @@ private function loadDifferentModelsFromPostRequest()
             }
 
             $modelPrototype->loadMultiple($models, $modelsData);
-            $result = array_merge($result, $models);
+            $result += $models;
         }
 
         return $result;
     }
+
+    private function ensureProgressive(array $models): void
+    {
+        $data = $this->request->post((new Threshold())->formName());
+        foreach ($models as $priceRowIdx => $model) {
+            if (!$model->isProgressive()) {
+                continue;
+            }
+            if (isset($data[$priceRowIdx])) {
+                $model->loadProgressiveData($data[$priceRowIdx]);
+            } else {
+                $model->class = 'SinglePrice';
```

Comment:
And if the class is equal to TemplatePrice, will you overwrite it with SinglePrice?